### ISSUE TITLE
Node Red crash without stacktrace if using watch node (23-watch.js)

### ISSUE THREAD

**Issue by lost-bits** [node-red/node-red#3355]
### Current Behavior

If you use the build in node "watch" (aka 23-watch.js) to monitor folders, it can cause node RED to crash like this:
![image](https://user-images.githubusercontent.com/75374180/149530861-e23acb82-5eed-4346-b028-635f572e03f7.png)

In the flow you can not catch the error by an catch node. Also like posted, even with "trace" in logging, you can't see the reason for the crash.

Remark:
Issue was originally posted at the [forum](https://discourse.nodered.org/t/strange-issue-with-reading-files-cyclic-from-a-local-drive-on-windows-error-ebusy-resource-busy-or-locked-lstat-c-hiberfil-sys-maybe-red-uncaught-exception/55336/8)

### Expected Behavior

If caused by the node (which I think is not the real reason, since 3rd party node [watchdirectory](https://github.com/fatoldsun00/node-red-contrib-watchdirectory) is causing the same crash), it should handle the error by catching it and throwing a error, the flow can handle.

Anyway node red should 1) not stop and 2) if stopped, post a better trace.

I really hope, you have an idea, what is happening.

### Steps To Reproduce

1. make sure, you have the following directories
c:\dir1
c:\dir2
2. make sure, you do not have the following directories (if they exist, please delete them or chance dirs in flow accordingly)
c:\dir3
c:\dir4
3.  Create a flow like below
4. Deploy the flow
5. Wait for aprox 30s to allow the watchers to init
6. create C:\dir3 or C:\dir4
7. NodeRed will crash

### Example flow

```
[{"id":"809a302f9bb08e6c","type":"tab","label":"Demonstrator Error","disabled":false,"info":"","env":[]},{"id":"777695dcd442bb18","type":"comment","z":"809a302f9bb08e6c","name":"how to reprodurce the following error","info":"Make sure you dont have only dir 1 and 2 (if 3 and 4 exist, please erase BEFORE you start nodered)\n\nC:\\Dir1\nC:\\Dir2\nC:\\Dir3\nC:\\Dir4\n\nSetup the Watchers to the dirs\n\nDeploy flow\n\nWait aprox 30s\n\nCreate \"dir3\" or \"dir4\" ... it will crash","x":190,"y":80,"wires":[]},{"id":"184a9ce7da6260b4","type":"watch","z":"809a302f9bb08e6c","name":"watch dir1","files":"c:\\\\dir1","recursive":"","x":180,"y":160,"wires":[["e502bca0f70fdb41"]]},{"id":"478352a49749057c","type":"watch-directory","z":"809a302f9bb08e6c","folder":"c:\\dir2","recursive":0,"typeEvent":"create","ignoreInitial":true,"ignoredFiles":"","ignoredFilesType":"re","name":"watch-directory dir2","x":210,"y":260,"wires":[["e502bca0f70fdb41"]]},{"id":"31392672ac1aeaef","type":"watch","z":"809a302f9bb08e6c","name":"watch dir3","files":"c:\\\\dir3","recursive":"","x":180,"y":200,"wires":[["e502bca0f70fdb41"]]},{"id":"9ea3790711e036f7","type":"watch-directory","z":"809a302f9bb08e6c","folder":"c:\\dir4","recursive":0,"typeEvent":"create","ignoreInitial":true,"ignoredFiles":"","ignoredFilesType":"re","name":"watch-directory dir4","x":210,"y":320,"wires":[["e502bca0f70fdb41"]]},{"id":"e502bca0f70fdb41","type":"debug","z":"809a302f9bb08e6c","name":"","active":true,"tosidebar":true,"console":true,"tostatus":true,"complete":"true","targetType":"full","statusVal":"payload","statusType":"auto","x":550,"y":220,"wires":[]},{"id":"af191e3ba6bb08a1","type":"catch","z":"809a302f9bb08e6c","name":"","scope":null,"uncaught":false,"x":470,"y":360,"wires":[["e502bca0f70fdb41"]]}]
```
![image](https://user-images.githubusercontent.com/75374180/149531815-4754ba53-8438-4bf9-b6c7-8ac7f2eea2f3.png)

### Environment

``- Node-RED version: 2.1.3 (also 1.3.4)
- Node.js version: 14.16.0 (also 12.21.0)
- npm version: 6.14.11
- Platform/OS: x64/Windows 10 #1909 build 18363.1977 (same problem on older machine)
- Browser: Edge 96.0.1054.57 (Same on Chrome)

- NodeRED ran by commandline:
```
@ECHO OFF
ECHO Starte NodeRED lokal - alternativ muss es als Dienst per NSSM konfiguriert werden
ECHO Set No TLS Erros (Self Signed Crt)
SET NOED_TLS_REJECT_UNAUTHORIZED=0
SET NO_UPDATE_notifier=1
ECHO Start Node
node.exe node_modules/node-red/red.js --settings=C:\\NodeRED.Temp2\\node-red_home\\settings.js
pause
```

**Comment by lost-bits** [comment]
As a workaround I automaticly restart nodered after the crash (same like a service) ... Anyway node red should catch the error and continue working.

```
@ECHO OFF
ECHO Starte NodeRED lokal - alternativ muss es als Dienst per NSSM konfiguriert werden
ECHO Set No TLS Erros (Self Signed Crt)
SET NOED_TLS_REJECT_UNAUTHORIZED=0
SET NO_UPDATE_notifier=1
ECHO Start Node

:startInstall

node.exe node_modules/node-red/red.js --settings=C:\\NodeRED.Temp2\\node-red_home\\settings.js
ECHO Looks like nodered crashed ....
timeout /T 10 /nobreak 

goto startInstall

pause

```

**Comment by Steve-Mcl** [comment]
Hi @lost-bits I can confirm node-red does crash as you describe.

However, this is only due to the fact node `node-red-contrib-watchdirectory` is installed.  
In fact, any new directory created in `c:\` causes this (_try creating a new directory named `fred`_) 
Additionally, even renaming any existing directory causes the issue.
Lastly, as proof, if you remove the `node-red-contrib-watchdirectory` nodes, then this problem does not occur. 


### Additional info

This occurs due to line 66 in `C:\Users\xxx\.node-red\node_modules\node-red-contrib-watchdirectory\node-red-watchdirectory.js` ...

```
      node.status({fill:"red", shape: "dot", text: "Error : "+err.message})
      throw(err)
```

The error is thrown in an `on` "error" handler that (due to scope) makes the error uncatchable in node-red.

As a test / temporary containment, you change that code to this...

```
      node.status({fill:"red", shape: "dot", text: "Error : "+err.message})
      //throw(err)
      node.error(err);
```
... then node-red is not crashed - but instead an entry in the debug is made ...

![image](https://user-images.githubusercontent.com/44235289/150753404-82f3a9dc-c5f2-4d99-ac6f-03a3c04306e7.png)

... though to be honest, I am not sure this would be a final fix (perhaps these unrelated errors should be simply ignored as they are not related to the watched directory)

---

In short, the way `node-red-contrib-watchdirectory` is throwing an error from inside the on error handler, node-red has no means of catching it.  You will need to [raise an issue on the repository](https://github.com/fatoldsun00/node-red-contrib-watchdirectory/issues) and feel free to link it back to this post for reference.

**Comment by Steve-Mcl** [comment]
@lost-bits I am closing this issue as I hopefully have proved this is not a node-red issue but in fact an issue with  `node-red-contrib-watchdirectory` & should be raised on [that repo](https://github.com/fatoldsun00/node-red-contrib-watchdirectory/issues)

If you believe otherwise, please re-open this issue.

Thanks for taking the time to report this.

**Comment by lost-bits** [comment]
Okay, thanks ... I will try removing the watcher from the nodered modules and test. If error remains, I will ping here.

**Comment by ajay396301** [comment]
but removing the watcher( watch directory) is not a solution. my project is the same as above I am also facing the same issue .how to fix the problem. while restarting the shared folder( network PC ) PC that time also same error.

**Comment by Steve-Mcl** [comment]
@ajay396301 

> but removing the watcher( watch directory) is not a solution

Unfortunately no, it is not, but it does demonstrates that this is not a `node-red` issue but instead an issue with the `node-red-contrib-watchdirectory` node.

You will need to raise an issue the the [node-red-contrib-watchdirectory repository](https://github.com/fatoldsun00/node-red-contrib-watchdirectory/issues) or fork that node and fix it.

**Comment by lost-bits** [comment]
yes, @Steve-Mcl  you are right. Since NodeRED WatchNode does not work on networkdrives (it does not see the changes somehow), I must live with the error or fix it ... and as said, if I find the error stays with node-red node, I will ping here. Thx so far.

**Comment by ajay396301** [comment]
please suggest how to fix it and what is the alternative ways to take data from network drive.

**Comment by ajay396301** [comment]
> yes, @Steve-Mcl you are right. Since NodeRED WatchNode does not work on networkdrives (it does not see the changes somehow), I must live with the error or fix it ... and as said, if I find the error stays with node-red node, I will ping here. Thx so far.

kindly suggest

**Comment by lost-bits** [comment]
@ajay396301  we should discuss the fix (if I find one) in ([node-red-contrib-watchdirectory repository](https://github.com/fatoldsun00/node-red-contrib-watchdirectory/issues)) as @Steve-Mcl mentioned.
